Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -461,21 +432,16 @@ open class ChartDataSet: ChartBaseDataSet
     {
         var removed = false
         
-        for i in 0 ..< _values.count
+        for i in 0 ..< values.count
         {
-            if _values[i] === entry
+            if values[i] === entry
             {
-                _values.remove(at: i)
+                values.remove(at: i)
                 removed = true
                 break
             }
         }
-        
-        if removed
```

Comment:
I don't see this can be deleted?

Reply:
Because `values` has a property observer that takes care of this now.

Reply:
Sorry, did this observer already exist? Or new added? I don't see it.

Reply:
It already existed.

Reply:
the last commit seems just reflect the change about count/endIndex.

Reply:
It was on `values`, not `_values`. It calls `notifyDataSetChanged()` which calls `calcMinMax()` and only `calcMinMax`

Reply:
why I feel the logic is different then. The old method will trigger `calcMinMax()` after something removed from _values. However in the new one, it does not call `calcMinMax()` if remove happens.
`values.remove(at: i)` will not call `notifyDataSetChanged()` I assume, it's not a setter call.

Reply:
It is a setter call. Try running this in a playground, you will see that the print statement is called.
```
var x = [1, 2, 3, 4] {
    didSet {
        print("ITEM REMOVED")
    }
}
x.remove(at: 1)
```

Reply:
what... that changed my understanding

Reply:
Is this behavior documented? I walked through https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Properties.html, not mentioning this specifically.

Reply:
In directly, yes. 
>If you pass a property that has observers to a function as an in-out parameter, the willSet and didSet observers are always called. This is because of the copy-in copy-out memory model for in-out parameters: The value is always written back to the property at the end of the function

Modification of array are all in-out (otherwise every time you modified an array, new memory would need to be allocated).

Reply:
Jesus, thanks. All good